src/class-block-converter.php

```diff
@@ -0,0 +1,387 @@
+
+		return new Block(
+			block_name: 'html',
+			content: $html,
+		);
+	}
+
+	/**
+	 * Get nodes from a specific tag.
+	 *
+	 * @param DOMNode $node The current DOMNode.
+	 * @param string  $tag The tag to search for.
+	 * @return DOMNodeList The raw HTML.
+	 */
+	public static function get_nodes( DOMNode $node, $tag ) {
+		return static::get_node_tag_from_html(
+			static::get_node_html( $node ),
+			$tag
+		);
+	}
+
+	/**
+	 * Get the raw HTML from a DOMNode node.
+	 *
+	 * @param DOMNode $node The current DOMNode.
+	 * @return string The raw HTML.
+	 */
+	public static function get_node_html( DOMNode $node ): string {
+		return $node->ownerDocument->saveHTML( $node );
+	}
+
+	/**
+	 * Get the HTML content.
+	 *
+	 * @param string $html The HTML content.
+	 * @param string $tag The tag to search for.
+	 * @return DOMNodeList The list of DOMNodes.
+	 */
+	public static function get_node_tag_from_html( $html, $tag = 'body' ) {
+		$dom = new \DOMDocument();
+
+		$errors = libxml_use_internal_errors( true );
+
+		$dom->loadHTML( '<?xml encoding="utf-8" ?>' . $html );
+
+		libxml_use_internal_errors( $errors );
+
+		return $dom->getElementsByTagName( $tag );
+	}
+
+	/**
+	 * Removing whitespace between blocks
+	 *
+	 * @param string $block Gutenberg blocks.
+	 * @return string
+	 */
+	protected function minify_block( $block ) {
+		if ( preg_match( '/(\s){2,}/s', $block ) === 1 ) {
+			return preg_replace( '/(\s){2,}/s', '', $block );
+		}
+
+		return $block;
+	}
+
+	/**
+	 * Quick way to remove all URL arguments.
+	 *
+	 * @param string $url URL.
+	 * @return string
+	 */
+	public function remove_image_args( $url ): string {
+		// Split url.
+		$url_parts = wp_parse_url( $url );
+
+		return $url_parts['scheme'] . '://' . $url_parts['host'] . $url_parts['path'];
+	}
+
+	/**
+	 * Upload image.
+	 *
+	 * @param string $src Image url.
+	 * @param string $alt Image alt.
+	 * @return string
+	 */
+	public function upload_image( string $src, string $alt ): string {
+		// Remove all image arguments.
+		$src = $this->remove_image_args( $src );
+
+		return create_or_get_attachment_from_url( $src, [ 'alt' => $alt ] );
+	}
+
+	/**
+	 * Remove any empty blocks.
+	 *
+	 * @param string $html The current HTML.
+	 * @return string $html The new HTML.
+	 */
+	public function remove_empty_blocks( string $html ): string {
+		$html = str_replace(
+			[
+// phpcs:disable
+'<!-- wp:html -->
+<div></div>
+<!-- /wp:html -->',
+'<!-- wp:paragraph -->
+<div> </div>
+<!-- /wp:paragraph -->',
+'<!-- wp:html -->
+<div> </div>
+<!-- /wp:html -->',
+'<!-- wp:paragraph -->
+<div>  </div>
+<!-- /wp:paragraph -->',
+'<!-- wp:paragraph --><p><br></p><!-- /wp:paragraph -->',
+'<!-- wp:paragraph --><p><br><br><br></p><!-- /wp:paragraph -->',
+'<!-- wp:paragraph -->
+<p><br></p>
+<!-- /wp:paragraph -->',
+'<!-- wp:html -->
+<div> </div>
+<!-- /wp:html -->',
+'<!-- wp:heading {"level":3} -->
+<h3>
+                                                        </h3>
+<!-- /wp:heading -->',
```

Comment:
I feel like there should be a better way to do this, but good enough for now...

Reply:
Yeah... I didn't write this part 🙉